app/models/electrical.rb

```diff
@@ -0,0 +1,43 @@
+class Electrical < ActiveRecord::Base
+  has_paper_trail class_name: 'Version'
+
+  belongs_to :country, required:false
+  has_one :package, as: :detail, dependent: :destroy
+  before_save :set_tested_on, if: :test_status_changed?
+  before_save :downcase_brand, if: :brand_changed?
+  before_save :set_updated_by
+  after_create :create_on_stockit
+  after_update :update_on_stockit
+
+  private
+
+  def set_tested_on
+    self.tested_on = Date.today
+  end
+
+  def downcase_brand
+    self.brand.downcase!
+  end
+
+  def create_on_stockit
+    response = Stockit::ItemDetailSync.create(self)
+    if response && (errors = response["errors"]).present?
+      errors.each { |key, value| self.errors.add(key, value) }
+    elsif response && (electrical_id = response["electrical_id"]).present?
+      self.update_column(:stockit_id, electrical_id)
```

Comment:
**NBD**: Use of extra variables is not required here since we are not calling any method, we are just using hash value.

```
    if response && response["errors"].present?
      response["errors"].each { |key, value| self.errors.add(key, value) }
    elsif response && response["electrical_id"].present?
      self.update_column(:stockit_id, response["electrical_id"])
```

Reply:
yeah right @swatijadhav , as @patrixr suggested, i'll create a concern and move the there and will remove the unwanted variables.